asserts/account_key.go

```diff
@@ -96,6 +104,21 @@ func (ak *AccountKey) checkConsistency(db RODatabase, acck *AccountKey) error {
 	if err != nil {
 		return err
 	}
+	// XXX: Make this unconditional once account-key assertions are required to have a name.
+	if ak.Name() != "" {
+		assertions, err := db.FindMany(AccountKeyType, map[string]string{
```

Comment:
Agreed.  It might be worth fixing in snap-assertions-service, which could apply a constraint in its PostgreSQL schema.  I don't think it's very urgent/likely in practice, as the only possible races would be for the same account.